modules/mixcr/1.2/mixcr.smk

```diff
@@ -0,0 +1,320 @@
+#!/usr/bin/env snakemake
+
+
+##### ATTRIBUTION #####
+
+
+# Original Author:  Anita dos Santos
+# Module Author:    Manuela Cruz
+# Contributors:     Laura Hilton
+
+
+##### SETUP #####
+
+
+# Import package with useful functions for developing analysis modules
+import oncopipe as op
+
+# Check that the oncopipe dependency is up-to-date. Add all the following lines to any module that uses new features in oncopipe
+min_oncopipe_version="1.0.11"
+import pkg_resources
+try:
+    from packaging import version
+except ModuleNotFoundError:
+    sys.exit("The packaging module dependency is missing. Please install it ('pip install packaging') and ensure you are using the most up-to-date oncopipe version")
+
+# To avoid this we need to add the "packaging" module as a dependency for LCR-modules or oncopipe
+
+current_version = pkg_resources.get_distribution("oncopipe").version
+if version.parse(current_version) < version.parse(min_oncopipe_version):
+    logger.warning(
+                '\x1b[0;31;40m' + f'ERROR: oncopipe version installed: {current_version}'
+                "\n" f"ERROR: This module requires oncopipe version >= {min_oncopipe_version}. Please update oncopipe in your environment" + '\x1b[0m'
+                )
+    sys.exit("Instructions for updating to the current version of oncopipe are available at https://lcr-modules.readthedocs.io/en/latest/ (use option 2)")
+
+# End of dependency checking section 
+
+# Setup module and store module-specific configuration in `CFG`
+# `CFG` is a shortcut to `config["lcr-modules"]["mixcr"]`
+CFG = op.setup_module(
+    name = "mixcr",
+    version = "1.2",
+    subdirectories = ["inputs", "mixcr", "outputs"],
+)
+
+assert type(CFG["igblastn"])==bool, (
+    "Ensure 'igblastn' is set to either True or False in config. "
+    "True: Runs igblastn alignment to retrieve % identity to IMGT sequences. This requires the '--contig-assembly' parameter when running MiXCR and may increase running time. "
+    )
+
+run_parameters = CFG["options"]["mixcr_run"]["mrna"]
+
+if CFG["igblastn"]:
+    assert "--contig-assembly" in run_parameters, (
+        "Config 'igblastn' value is set to True, but ['mixcr_run']['options'] does not include '--contig-assembly'. "
+        "Include '--contig-assembly' in CFG['mixcr_run']['options'] to run igblastn.\n "
+        "\n***** NOTE: using '--contig-assembly' may increase run duration. *****\n"
+    )
+
+RECEPTORS = CFG["receptors"]
+
+RANGE = ["IGH","IGL","IGK","TRA","TRB","TRD","TRG"]
+
+ig_type = ["IGH","IGK","IGL"]
+tr_type = ["TRA", "TRB", "TRD", "TRG"]
+
+if isinstance(RECEPTORS, str):
+    RECEPTORS = RECEPTORS.split(" ")
+if isinstance(RECEPTORS, list) and len(RECEPTORS) == 1:
```

Comment:
What about the case where `RECEPTORS` is a list of length > 1?

Reply:
Hmmm here I am assuming that if the length is longer than > 1, that the receptor input is multiple receptors for example ["IGH","IGK"] or ["TRA","TRB"]. The assert statement below makes sure that all receptors specified are within the accepted values, so I think that will handle the cases where len(RECEPTORS) > 1 without having to handle them in this block of code